### PR TITLE
on 'number of bids' click dialog with auction bids history opens

### PR DESCRIPTION
**Pull request by draganaaz**
- Created getAuctionBids endpoint
- On click 'number of bids' a dialog with table opens showing auction bids (users' bids history)
- Updated datatable component with necessary columns (because we use one single mat-table in ecvery component)
- Added getAuctionBids(id) method in auctionService
- Formatted time from database
- Updated datetime format in datatable details component
- Updated name of service in auction service test file from AuctionFetchService to AuctionService
- Also, I hardcoded auctionID on both, frontend and backend since we still don't have bidding enabled; once we implement that, it will be updated.

### CODE REVIEW

File: auction9-backend/serverless.yml [hunk 1]
```diff
@@ -57,7 +57,16 @@ functions:
           method: post
           cors: true
 
-  # return all my auctions
+  # returns auction bids (users' bids history)
+  getAuctionBids:
+    handler: handler.getAuctionBids
+    events:
+      - http:
+          path: auctionBids/{auctionID}
```
Comment (mjovanovic) [inline auction9-backend/serverless.yml:65]: Let's change the path to `auctions/{id}/bids

File: auction9-frontend/src/app/data-table/data-table-details/data-table-details.component.html [hunk 2]
```diff
@@ -23,22 +25,31 @@ <h4>Status:
...
     <p>
       {{auction?.description}}
     </p>
   </mat-card-content>
   <mat-card-actions>
   	<mat-form-field class="input-bid">
-	    <input matInput placeholder="Offer must start from: {{auction?.price + 1}} din" value="{{auction?.price + 1}}" min="{{auction?.price + 1}}" required>
+      <input
+        matInput
+        placeholder="Offer must start from: {{auction?.price + 1}} din"
```
Comment (mjovanovic) [inline auction9-frontend/src/app/data-table/data-table-details/data-table-details.component.html:48]: We should make the inncrement configurable. I'll create a ticket

File: auction9-frontend/src/app/history-dialog/history-dialog.component.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+})
+export class HistoryDialogComponent implements OnInit {
+
+  tableData: any[];
+  tableHeaders: string[];
+
+  constructor(private auctionService: AuctionService) { }
+
+  ngOnInit(): void {
+    // TODO: once bidding is enabled, pass the 'auctionID' instead of hardcoded '1'
```
Comment (mjovanovic) [inline auction9-frontend/src/app/history-dialog/history-dialog.component.ts:17]: You can already pass this as an argument to this component

File: auction9-backend/serverless.yml (hunk 1 above)
Comment (draganaaz) [reply auction9-backend/serverless.yml:65]: Sure, I'll change it.

File: auction9-frontend/src/app/data-table/data-table-details/data-table-details.component.html (hunk 2 above)
Comment (draganaaz) [reply auction9-frontend/src/app/data-table/data-table-details/data-table-details.component.html:48]: Stefan said he will take care of that, okay.

File: auction9-backend/handler.js [hunk 4]
```diff
@@ -87,6 +87,25 @@ export const postAuction = async (event, context) => {
...
 };
 
+/* getAuctionBids - returns auction bids (users' bids history)
+ * GET: auctions/{id}/bids
+ */
+export const getAuctionBids = async (event, context) => {
+  try {
+    // once we enable bidding, remove hardcoded auctionID
+    // let resultAuctionBids = await mysql.query('SELECT u.name, price, time FROM tbl_user_auction ua JOIN tbl_user u on (ua.userID = u.userID) WHERE auctionID=? ORDER BY price', [auctionID]);
+    let resultAuctionBids = await mysql.query('SELECT u.name, price, time FROM tbl_user_auction ua JOIN tbl_user u on (ua.userID = u.userID) WHERE auctionID=1 ORDER BY price');
```
Comment (SvetislavMarjanovic) [inline auction9-backend/handler.js:97]: Are we sure that we want to use userId in this query and to do join per user. I think that this query should return all bids no matter on user id, only actionId is required.

File: auction9-frontend/src/app/services/auction.service.ts [hunk 5]
```diff
@@ -11,37 +11,44 @@ export class AuctionService {
...
+    return this.http.get(`${environment.baseUrl}/auctions/${auctionID}/bids`, { responseType: 'json' }).toPromise();
+  }
+
+  /* getMyAuctions - return all auctions for current user
+  * Method: GET
+  * Path: /myauctions
+  */
+  // currently userId hard coded for testing purpose
+  // after SSO is implemented, this will be updated
   getMyAuctions() {
```
Comment (SvetislavMarjanovic) [inline auction9-frontend/src/app/services/auction.service.ts:49]: Name of this method should be getUserAuctions, same goes for lambda name in serverless file and path.

Comment (draganaaz) [reply auction9-frontend/src/app/services/auction.service.ts:49]: Okay.

**Pull request merged**